tools/cli/main.cpp

```diff
@@ -19,6 +19,46 @@
 #include <tuple>
 #include <sys/wait.h>
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#include <cstdlib>
+#include <cerrno>
+#include <cstring>
+#include <unistd.h>
+inline char** get_environ() {
+    return *_NSGetEnviron();
+}
+constexpr const char* AGENT_LIBRARY = "libbpftime-agent.dylib";
+constexpr const char* SYSCALL_SERVER_LIBRARY = "libbpftime-syscall-server.dylib";
+constexpr const char* AGENT_TRANSFORMER_LIBRARY = "libbpftime-agent-transformer.dylib";
+const char *strchrnul(const char *s, int c) {
+    while (*s && *s != (char)c) {
+        s++;
+    }
+    return s;
+}
+int execvpe(const char *file, char *const argv[], char *const envp[]) {
+    for (const char *path = getenv("PATH"); path && *path; path = strchr(path, ':') + 1) {
+        char buf[PATH_MAX];
+        const char *end = strchrnul(path, ':');
+        size_t len = end - path;
+        memcpy(buf, path, len);
+        buf[len] = '/';
+        strcpy(buf + len + 1, file);
+        execve(buf, argv, envp);
+        if (errno != ENOENT)
+            return -1;
+    }
+    errno = ENOENT;
+    return -1;
+}
+#else
```

Comment:
Can we do `#elif __linux__` so it fails on other platforms that haven't been implemented?